### DISCUSSION TITLE
Incorrect behavior for column labeling

### DISCUSSION THREAD

**Discussion by reuvenstr** [sqlalchemy/sqlalchemy#11863] [Usage Questions] [answered]
Hi,

During migration to 2.0, I faced an issue with column labeling, that currently breaks the core of my query abstraction. 
Can you please confirm that it's the desired behavior, and just in the previous version like 1.3 it was not used correctly from my side? Thanks

Steps to reproduce

Using sqlalchemy 1.3 output of print will be `name`, when using version 2.0 then the output is `anon_1.name`
```
from sqlalchemy import create_engine, Column, Integer, String
from sqlalchemy.ext.declarative import declarative_base
from sqlalchemy.orm import sessionmaker

# Define the database URL (using SQLite in-memory database for simplicity)
DATABASE_URL = "sqlite:///:memory:"

# Create an engine
engine = create_engine(DATABASE_URL, echo=True)

# Define a base class for the ORM models
Base = declarative_base()


# Define a table/model
class User(Base):
    __tablename__ = 'users'

    id = Column(Integer, primary_key=True)
    name = Column(String)
    age = Column(Integer)


# Create the table in the database
Base.metadata.create_all(engine)

# Create a session
Session = sessionmaker(bind=engine)
session = Session()

# Perform a SELECT query using SQLAlchemy 1.4 syntax
print(session.query(User).statement.c['name'])
```

**Comment by CaselIT** [comment]
Hi,

Please refer to the documentation https://docs.sqlalchemy.org/en/20/core/selectable.html#sqlalchemy.sql.expression.SelectBase.c

**Reply by reuvenstr** [reply]
> Hi,
> 
> Please refer to the documentation https://docs.sqlalchemy.org/en/20/core/selectable.html#sqlalchemy.sql.expression.SelectBase.c

Thanks for reply!
You are talking about deprecation or that subquery should  be invoked during such use?
I didn’t find anything related to column labeling change..

**Reply by CaselIT** [reply, marked as answer]
Subquery is automatically called by `.c`. without an explicit name the subquery uses `anon_n` as name

**Reply by reuvenstr** [reply]
Thats what I thought, make more sense now. Are you familiar if there is some method with same behavior that was before 1.4?

**Reply by reuvenstr** [reply]
Nevermind, there is a recommendation to use `selected_columns`. It failed my test, and I didn't check the cause. 😃 
Thanks a lot

**Reply by CaselIT** [reply]
again refer to the documentation. it suggest using `selected_columns`

**Comment by reuvenstr** [comment]
In order to limit the number of open discussions, I would like to ask here.
I faced something strange, statement that was extracted from query.options(load_only) show different results, there is some explanation in the docs about it?

Using sqlalchemy 1.3 output of print will be:
```
SELECT users.id AS users_id
FROM users
['id']
```

when using version 1.4 then the output, in 2.0 it's resuires to use column instead of str:
```
SELECT users.id AS users_id
FROM users
ImmutableColumnCollection(users.id, users.name, users.age)
```

```
from sqlalchemy import create_engine, Column, Integer, String
from sqlalchemy.ext.declarative import declarative_base
from sqlalchemy.orm import sessionmaker

# Define the database URL (using SQLite in-memory database for simplicity)
DATABASE_URL = "sqlite:///:memory:"

# Create an engine
engine = create_engine(DATABASE_URL, echo=True)

# Define a base class for the ORM models
Base = declarative_base()


# Define a table/model
class User(Base):
    __tablename__ = 'users'

    id = Column(Integer, primary_key=True)
    name = Column(String)
    age = Column(Integer)


# Create the table in the database
Base.metadata.create_all(engine)

# Create a session
Session = sessionmaker(bind=engine)
session = Session()

# check select
print(str(session.query(User).options(load_only(*['id']))))
print(session.query(User).options(load_only(*['id']).selected_columns) # in 1.3 selected_columns changed to c
```

**Reply by CaselIT** [reply]
I don't think the loading options are applied to the select yet.

Also note that in 1.4+ you can use `select(User)` instead of query.. statement

**Reply by reuvenstr** [reply]
It actually has the same behavior, to use select or query, the difference is that using query adds to the object session attribute..
